GVFS/GVFS.Notifications/VFSForGit.Mac/StatusMenuItem/IPC/VFSForGitNotification.m

```diff
@@ -0,0 +1,183 @@
+#import "VFSForGitNotification.h"
+
+NSString * const IdentifierKey = @"Id";
+NSString * const EnlistmentKey = @"Enlistment";
+NSString * const EnlistmentCountKey = @"EnlistmentCount";
+NSString * const TitleKey = @"Title";
+NSString * const MessageKey = @"Message";
+
+NSString * const AutomountTitle = @"GVFS AutoMount";
+NSString * const AutomountStartMessageFormat = @"Attempting to mount %lu GVFS repos(s)";
+NSString * const AutomountSuccessMessageFormat = @"The following GVFS repo is now mounted: \n%@";
+NSString * const AutomountFailureMessageFormat = @"The following GVFS repo failed to mount: \n%@";
+
+@interface VFSForGitNotification()
+
+@property (readwrite) NSString *title;
+@property (readwrite) NSString *message;
+
+NS_ASSUME_NONNULL_BEGIN
+
+- (instancetype _Nullable)initAsMountSuccessWithMessage:(NSDictionary *)messageDict
+                                                  error:(NSError *__autoreleasing *)error;
+
+- (instancetype _Nullable)initAsMountFailureWithMessage:(NSDictionary *)messageDict
+                                                  error:(NSError *__autoreleasing *)error;
+
+- (instancetype _Nullable)initAsMountWithMessage:(NSDictionary *)messageDict
+                                           title:(NSString *)title
+                                   messageFormat:(NSString *)messageFormat
+                                           error:(NSError *__autoreleasing *)error;
+
+NS_ASSUME_NONNULL_END
+
+@end
+
+@implementation VFSForGitNotification
+
++ (BOOL)tryValidateMessage:(NSDictionary *)jsonMessage
+         buildNotification:(VFSForGitNotification **)notification
+                     error:(NSError *__autoreleasing *)error
+{
+    NSParameterAssert(notification);
+    NSParameterAssert(jsonMessage);
+    
+    id identifier = jsonMessage[IdentifierKey];
+    if (![identifier isKindOfClass:[NSNumber class]])
+    {
+        if (error != nil)
+        {
+            *error = [NSError errorWithDomain:@"VFS For Git"
+                                         code:-1
```

Comment:
Suggestion: Ideally each error code should be unique to identify the failure. Helps when you want the caller to handle particular error conditions (by comparing error code numbers/enums).

Reply:
Added Notification specific error codes in `VFSNotificationErrors.h`